hadoop-mapreduce/src/main/java/org/apache/accumulo/hadoop/mapreduce/OutputInfo.java

```diff
@@ -0,0 +1,147 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.hadoop.mapreduce;
+
+import java.util.Optional;
+import java.util.Properties;
+
+import org.apache.accumulo.core.client.BatchWriter;
+import org.apache.accumulo.core.client.BatchWriterConfig;
+import org.apache.accumulo.core.client.ClientInfo;
+import org.apache.accumulo.hadoopImpl.mapreduce.OutputInfoImpl;
+import org.apache.hadoop.mapreduce.Job;
+
+/**
+ * Object containing all the information needed for the Map Reduce job. This object is passed to
+ * {@link AccumuloOutputFormat#setInfo(Job, OutputInfo)}. It uses a fluent API like so:
+ *
+ * <pre>
+ * OutputInfo.builder().clientInfo(clientInfo).batchWriterOptions(bwConfig).build();
+ * </pre>
+ *
+ * @since 2.0
+ */
+public interface OutputInfo {
+  /**
+   * @return the client info set using OutputInfo.builder().clientInfo(info)
+   */
+  ClientInfo getClientInfo();
+
+  /**
+   * @return the client properties set using OutputInfo.builder().clientInfo(info)
+   */
+  Properties getClientProperties();
+
+  /**
+   * @return the BatchWriterConfig set using OutputInfo.builder()...batchWriterOptions(conf)
+   */
+  Optional<BatchWriterConfig> getBatchWriterOptions();
+
+  /**
+   * @return the default tame name set using OutputInfo.builder()...defaultTableName(name)
+   */
+  Optional<String> getDefaultTableName();
+
+  /**
+   * @return boolean if creating tables or not
+   */
+  boolean isCreateTables();
+
+  /**
+   * @return boolean if running simulation mode or not
+   */
+  boolean isSimulationMode();
+
+  /**
+   * @return builder for creating a {@link OutputInfo}
+   */
+  public static OutputInfoBuilder.ClientParams builder() {
+    return new OutputInfoImpl.OutputInfoBuilderImpl();
+  }
+
+  /**
+   * Fluent API builder for OutputInfo
+   *
+   * @since 2.0
+   */
+  interface OutputInfoBuilder {
+
+    /**
+     * Required params for client
+     *
+     * @since 2.0
+     */
+    interface ClientParams {
+      /**
+       * Set the connection information needed to communicate with Accumulo in this job. ClientInfo
+       * param can be created using {@link ClientInfo#from(String)} or
+       * {@link ClientInfo#from(Properties)}
+       *
+       * @param clientInfo
+       *          Accumulo connection information
+       */
+      OutputOptions clientInfo(ClientInfo clientInfo);
+    }
+
+    /**
+     * Builder options
+     *
+     * @since 2.0
+     */
+    interface OutputOptions {
+      /**
+       * Sets the configuration for for the job's {@link BatchWriter} instances. If not set, a new
+       * {@link BatchWriterConfig}, with sensible built-in defaults is used. Setting the
+       * configuration multiple times overwrites any previous configuration.
+       *
+       * @param bwConfig
+       *          the configuration for the {@link BatchWriter}
+       */
+      OutputOptions batchWriterOptions(BatchWriterConfig bwConfig);
```

Comment:
Users can set `BatchWriterConfig` when they use [the AccumuloClient builder](https://github.com/apache/accumulo/blob/master/core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java#L562) to build `ClientInfo`.  Therefore, this method could be removed.

Reply:
I think this is a good idea since we could reduce what goes into the API.  My only concern is that it will be confusing of which options should be set through which builder.

Reply:
It's up to you if you want to remove it, but if you keep it you should tell users what happens if they set the config in both place (i.e. tell them this method takes precedence over the method in AccumuloClient builder) and make sure this actually occurs in the implementation.

Reply:
With the current state of the code, it would take some effort under the hood to redo how the batch writer config is stored and serialized.  I think I will make this a follow on blocker.

Reply:
Created #751